src/wasm/wasm-type.cpp

```diff
@@ -43,49 +43,46 @@ size_t hash<wasm::Type>::operator()(const wasm::Type& type) const {
 }
 
 size_t hash<wasm::Signature>::operator()(const wasm::Signature& sig) const {
-  auto res = hash<uint64_t>{}(sig.params.getID());
-  wasm::hash_combine(res, sig.results.getID());
-  return res;
+  return wasm::rehash(uint64_t(hash<uint64_t>{}(sig.params.getID())),
+                      uint64_t(hash<uint64_t>{}(sig.results.getID())));
```

Comment:
I reverted this one to the old implementation before my PR as well. Here, the `.getID()`s are piped through `std::hash`, from `uint64_t` to `size_t` back to `uint64_t` to obtain the final 64-bit hash value then returned as a `size_t`.